frontend/src/app/_shared/services/products.service.ts

```diff
@@ -41,16 +41,20 @@ export class ProductsService {
    * @param {string} [searchTerm='']
    * @param {string} [sort='']        - Add ' ASC'/' DESC'
    * @param {number} [minPrice=0]
+   * @param {number} [maxPrice=Infinity]
    * @returns {Observable<ProductModel[]>}
    */
-  searchProducts(limit = 20, skip = 0, searchTerm = '', sort = '', minPrice = 0): Observable<ProductModel[]>{
+  searchProducts(limit = 20, skip = 0, searchTerm = '', sort = '', minPrice = 0, maxPrice = Infinity): Observable<ProductModel[]>{
     let where = {
       or: [
         { name: {contains: searchTerm} },
         { description: {contains: searchTerm} },
       ],
       listed: true, // We only want products that are still buyable
-      price: {">": minPrice},
+      price: {
+        ">=": minPrice,
+        "<=": (isFinite(maxPrice) ? maxPrice : 1000000000000), // Just toss in som giant number for inf
```

Comment:
May be better to send the request without the <= field if maxPrice is non finite.